cumulus/client/cirrus-executor/src/aux_schema.rs

```diff
@@ -51,29 +50,31 @@ pub(super) fn write_execution_receipt<Backend: AuxStore, Block: BlockT, PBlock:
 
 	let mut new_first_saved_receipt = first_saved_receipt;
 
-	let keys_to_delete = if block_number - first_saved_receipt >= PRUNING_DEPTH.saturated_into() {
-		new_first_saved_receipt = block_number.saturating_sub((PRUNING_DEPTH - 1).saturated_into());
-
-		let mut keys_to_delete = vec![];
-		let mut to_delete_start = first_saved_receipt;
-		while to_delete_start < new_first_saved_receipt {
-			let delete_block_number_key =
-				(EXECUTION_RECEIPT_BLOCK_NUMBER, to_delete_start).encode();
-			if let Some(hashes_to_delete) =
-				load_decode::<_, Vec<Block::Hash>>(backend, delete_block_number_key.as_slice())?
-			{
-				keys_to_delete.extend(
-					hashes_to_delete.into_iter().map(|h| (EXECUTION_RECEIPT_KEY, h).encode()),
-				);
-				keys_to_delete.push(delete_block_number_key);
+	let keys_to_delete =
+		if best_execution_chain_number >= first_saved_receipt + PRUNING_DEPTH.saturated_into() {
```

Comment:
>Again, this refactoring of replacing subtraction with addition is valid, but it is not a natural way of writing it. Why?

`block_number - first_saved_receipt` in our context is totally fine, but without any context, it has a subtract overflow risk in theory, just wanted to make the code look robust :P I can revert if you like.

>Also can't PRUNING_DEPTH.saturated_into() be replaced with PRUNING_DEPTH.into() here?

You're right, it can.

Reply:
As in the other comment you can use saturating subtraction. For types that support `.into()` it should be preferred, such that as soon as type can't be converted without data loss it stops compiling and complains.